app/models/concerns/permalinkable.rb

```diff
@@ -9,7 +9,15 @@ module Permalinkable
 
   module ClassMethods
     def find_by_param!(param)
-      find_by_permalink!(param)
+      begin
+        find_by_permalink!(param)
+      rescue ActiveRecord::RecordNotFound
+        if param =~ /^\d+$/
```

Comment:
If it doesn't have a big impact I'll see what I can do. This is being used everywhere we need to load a project before project-related controller actions are executed. ;)